lib/Dialect/AIE/Transforms/AIEObjectFifoStatefulTransform.cpp

```diff
@@ -1034,10 +1036,13 @@ struct AIEObjectFifoStatefulTransformPass
         }
         BDDimLayoutArrayAttr emptyDims =
             BDDimLayoutArrayAttr::get(builder.getContext(), {});
+        // Avoid overflow in case of a single BDDimLayoutAttr for multiple
+        // consumers.
+        ArrayRef<BDDimLayoutAttr> currentConsumerDim =
+            consumerIndex < consumerDims.size() ? consumerDims[consumerIndex]
+                                                : consumerDims[0];
```

Comment:
Instead of returning the same data layout transformation for all consumers, consumers without a data layout transformation should receive an empty list of dims.

Reply:
This is an indication that the op semantics are underspecified. Not surprising because broadcast semantics are annoyingly "subtle". But I would strongly advise to emulate a common set of semantics (eg numpy) rather than deciding from first principles. And in numpy it (I believe, will need to double check) ever consumer gets the same transformation (ie slice object ).

Reply:
I can understand that logic, however, it isn't the logic implemented by the custom ObjectFifoCreateOp. We can certainly discuss a change with a larger team internally and agree then on a new syntax that allows us to specify the patterns which we want to see and use across projects. If we do change it in the future as well then we should take the time to update the relevant doc (or add more if it is lacking).